rclpy/test/test_timer.py

```diff
@@ -168,19 +170,23 @@ def test_time_until_next_call() -> None:
         timer = node.create_timer(1, lambda: None)
         assert not timer.is_canceled()
         executor.spin_once(0.1)
-        assert timer.time_until_next_call() <= (1 * S_TO_NS)
+        time_until_next_call = timer.time_until_next_call()
+        assert time_until_next_call is not None
+        assert time_until_next_call <= (1 * S_TO_NS)
         timer.reset()
         assert not timer.is_canceled()
-        assert timer.time_until_next_call() <= (1 * S_TO_NS)
+        time_until_next_call = timer.time_until_next_call()
+        assert time_until_next_call is not None
+        assert time_until_next_call <= (1 * S_TO_NS)
         timer.cancel()
         assert timer.is_canceled()
         assert timer.time_until_next_call() is None
     finally:
-        if timer is not None:
-            node.destroy_timer(timer)
         if executor is not None:
             executor.shutdown()
         if node is not None:
+            if timer is not None:
+                node.destroy_timer(timer)
```

Comment:
although i understand this makes sure the node exists and then destroys the timer, did we really have any problems with previous code? just checking, no need to change.

Reply:
It was not type safe but, in practice the node was always there.